### PR TITLE
Auth APIs bugs resolved

### LINKED ISSUE #12: Auth APIs
* verify_email => return "already verified" instead of error if the db column is verified already
* forgot_password => check if the email is verified before sending the reset password link

### PR DESCRIPTION
**Pull request by ridhishjain**
resolves #12

### CODE REVIEW

File: server/models/auth/forgotPassword.js [hunk 1]
```diff
@@ -20,6 +20,24 @@ function forgotPassword({ email }) {
...
 
+    pool.query(
+      `SELECT verified FROM user WHERE email=?`,
+      [email],
+      (error, results) => {
+        if (error) {
+          return reject(error)
+        }
+        if (!results.length) {
+          return reject('No account exists with the given email!')
```
Comment (ridhishjain) [inline server/models/auth/forgotPassword.js:31]: if results here comes empty, means no email exists!

File: server/models/auth/forgotPassword.js [hunk 2]
```diff
@@ -20,6 +20,24 @@ function forgotPassword({ email }) {
...
+        }
+        const { verified } = results[0]
+        if (verified !== 1) {
+          return reject(
+            'The account with this email is not verified yet, please verify your email account first!'
```
Comment (ridhishjain) [inline server/models/auth/forgotPassword.js:36]: if email exists and not verified, it needs to get verified first

File: server/models/auth/verifyEmail.js [hunk 3]
```diff
@@ -11,6 +11,19 @@ const { pool } = require('../database')
...
   return new Promise((resolve, reject) => {
+    pool.query(
+      `SELECT verified FROM user WHERE username=?`,
+      [username],
+      (error, results) => {
+        if (error || !results.length) {
+          return reject(error)
+        }
+        const { verified } = results[0]
+        if (verified === 1) {
```
Comment (ridhishjain) [inline server/models/auth/verifyEmail.js:22]: if email is already verified

**Comment by ridhishjain** [comment]
@cjchirag7 review

**Comment by cjchirag7** [comment]
LGTM

File: server/models/auth/forgotPassword.js (hunk 1 above)
Comment (cjchirag7) [reply server/models/auth/forgotPassword.js:31]: Great !

File: server/models/auth/forgotPassword.js (hunk 2 above)
Comment (cjchirag7) [reply server/models/auth/forgotPassword.js:36]: Fine

**Review by cjchirag7 — APPROVED** [review]
Great ! We shall merge this

**Pull request merged**